### PR TITLE
chore(api): Update API dev deps for cross-platform Python 3.6.4

### PR DESCRIPTION
**Pull request by IanLondon**
(Edit: written by @mcous)

To prevent problems with [robot update](https://github.com/Opentrons/opentrons/milestone/20), we need to make sure that our development Python version and dependencies match production as closely as possible and that everything works cross-platform (because update files will be generated on multiple platforms).

The "cross-platform" part of things turned out to be terrible.

This PR threw a bunch of stuff at the wall to get stuff to stick:

- Updated pipenv to 11.6.8 for numerous bugfixes
- Updated pytest to 3.4.2 and locks coverage to 4.4.2 to fix weird pytest plugin bugs in Windows
- Refactored use of pipenv
    - Removed `--system` option to allow pipenv to create a virtualenv
    - Limited use of `--ignore-pipfile` option to CI
    - Updated `api/Makefile` to use the virtualenv
- Delegated prod dep versions to `setup.py` and lock them there
    - Also updated a few Python version callouts to `3.6`
- Generated `Pipfile.lock` on Ubuntu for safety
    - I spent a lot of time comparing lockfiles generated by pipenv@11.6 and windows
    - It appears the lockfiles are completely cross-platform now, so:
    - We probably don't need to do this moving forward 

Anyway, API `install`, `dev`, and `test` tasks are all passing on my macOS 10.12, Windows 10, and Ubuntu 14.04. Please verify on your machines as well:

1. Ensure you have an install of Python 3.6
    - *nix - `pyenv install 3.6.4`
    - Windows - `choco install python --version=3.6.4.20180116 --yes`
2. If you already had pipenv installed, do:
    1. `cd api`
    2. Remove any virtual env that may exist - `pipenv --rm`
    3. Uninstall pipenv for safety - `pip uninstall pipenv`
    4. `cd ..`
3. Run top level install task - `make install`
4. Run API level test `make test-api` or `make -C api test`
5. Run API level dev task - `ENABLE_VIRTUAL_SMOOTHIE=true make -C api dev`
    - Probably run the app dev task, too: `make -C app dev`

P.S. I'm going to squash this PR before merging

### CODE REVIEW

File: api/Pipfile.lock [hunk 1]
```diff
@@ -1,525 +1,523 @@
...
-    }
-}
+{
+    "_meta": {
+        "hash": {
+            "sha256": "037408de688057ac7651666dd911d64061da9c1078d6aa45f67eab1cf4b70a8f"
+        },
+        "host-environment-markers": {
+            "implementation_name": "cpython",
+            "implementation_version": "3.5.3",
```
Comment (mcous) [inline api/Pipfile.lock]: Can you regenerate this on Python 3.6.4?

Comment (IanLondon) [reply api/Pipfile.lock]: d'oh yup

**Comment by mcous** [comment]
`pipenv run` appears to be swallowing `SIGINT` on Windows?? Trying to build a simple repro

**Comment by mcous** [comment]
Been doing some research, and I actually can't get ctrl-C to work with `make dev` on Windows on `edge`...`make` quits but the process is left running. If I try running the process without `make` ctrl-C does nothing at all (same behavior with and without pipenv). @btmorr or @Laura-Danielle have you experienced this? Not only can I not get it to work, I can't even get the [aiohttp hello world](https://aiohttp.readthedocs.io/en/stable/web_quickstart.html) to quit with ctrl-C on Windows...

``` py
# server.py
# tested on python==2.6.4, aiohttp==2.3.8 and aiohttp==3.0.7
from aiohttp import web


async def hello(request):
    return web.Response(text="Hello, world")


def init(loop=None):
    app = web.Application()
    app.router.add_get('/', hello)
    return app


web.run_app(init(), host='localhost', port=8080)
```

This would seem to indicate that ctrl-C misbehavior on Windows is _not_ a regression introduced by this PR, so I would be inclined to merge and move forward

**Comment by mcous** [comment]
After talking with @btmorr, confirmed that ctrl-C problems on Windows are not a regression and likely an underlying issue with aiohttp or asyncio

**Pull request merged**